spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb

```diff
@@ -0,0 +1,97 @@
+require_domain_file
+
+describe ManageIQ::Automate::Service::Retirement::StateMachines::Methods::StartRetirement do
+  # describe "start_retirement Method Validation" do
+  let(:admin) { FactoryBot.create(:user_admin) }
+  let(:request) { FactoryBot.create(:service_retire_request, :requester => admin) }
+  let(:service) { FactoryBot.create(:service) }
+  let(:task) { FactoryBot.create(:service_retire_task, :destination => service, :miq_request => request) }
+  let(:svc_task) { MiqAeMethodService::MiqAeServiceServiceRetireTask.find(task.id) }
+  let(:svc_service) { MiqAeMethodService::MiqAeServiceService.find(service.id) }
+  let(:root_object) do
+    Spec::Support::MiqAeMockObject.new('service'             => svc_service,
+                                       'service_retire_task' => svc_task)
+    #   'service_action'      => 'Retirement')
+  end
+  let(:ae_service) do
+    Spec::Support::MiqAeMockService.new(root_object).tap do |service|
+      current_object = Spec::Support::MiqAeMockObject.new
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  let(:svc_model_request) do
+    MiqAeMethodService::MiqAeServiceServiceRetireRequest.find(request.id)
+  end
+
+  it "#start_retirement is nil and service is retiring" do
+    expect(service.retirement_state).to be_nil
+    service.start_retirement
+
+    expect(service.retirement_state).to eq("retiring")
+  end
+
+  it "#retiring - false" do
+    expect(service.retiring?).to be_falsey
+  end
+
+  it "#retiring? - true" do
+    service.retirement_state = 'retiring'
+
+    expect(service.retiring?).to be_truthy
+  end
+
+  it "#retired - false" do
+    expect(service.retired?).to be_falsey
+  end
+
+  it "#retired? - true" do
+    service.retirement_state = 'retired'
+    service.retired = true
+    expect(service.retired?).to be_truthy
+  end
+
+  it "starts retirement" do
```

Comment:
What does this test?

Reply:
This does not test anything.

Reply:
I'm trying to test the good condition.  Retirement_state is nil and then I start retirement and the retirement_state is equal to retiring.